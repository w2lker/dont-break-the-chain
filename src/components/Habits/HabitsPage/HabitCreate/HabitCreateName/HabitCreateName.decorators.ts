import withStyles from '@material-ui/core/styles/withStyles';
import HabitCreateName from './HabitCreateName';
import habitCreateNameStyles from './HabitCreateName.styles';

const styledComponent = withStyles(habitCreateNameStyles)(HabitCreateName);

export default styledComponent;
