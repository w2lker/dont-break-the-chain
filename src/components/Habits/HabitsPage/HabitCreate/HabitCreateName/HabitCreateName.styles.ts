const habitCreateNameStyles = {
  root: {
    padding: 0,
  },
};

export default habitCreateNameStyles;
