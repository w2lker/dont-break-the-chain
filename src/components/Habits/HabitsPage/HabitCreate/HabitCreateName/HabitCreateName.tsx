import React from 'react';

import { WithStyles } from '@material-ui/styles';

import habitCreateNameStyles from './HabitCreateName.styles';

export interface IHabitCreateNameProps extends WithStyles<typeof habitCreateNameStyles> {}

const HabitCreateName: React.FC<IHabitCreateNameProps> = (props) => {
  const { classes } = props;
  return (
    <div className={ classes.root }>
      HabitCreateName Component
    </div>
  );
};

export default HabitCreateName;
