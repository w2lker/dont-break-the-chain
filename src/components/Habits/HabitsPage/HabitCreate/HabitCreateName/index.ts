import HabitCreateName from './HabitCreateName.decorators';

export default HabitCreateName;
