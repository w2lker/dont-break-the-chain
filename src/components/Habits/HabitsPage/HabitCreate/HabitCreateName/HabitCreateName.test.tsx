import React from 'react';

import { mount, shallow } from 'enzyme';
// import cases from 'jest-in-case';

import HabitCreateName, { IHabitCreateNameProps } from './HabitCreateName';
import HabitCreateNameDecorated from './HabitCreateName.decorators';
import habitCreateNameStyles from './HabitCreateName.styles';

const testSetup = () => {
  const classes = {
    root: 'habitCreateName-wrapper',
  };
  const classesKeys = Object.keys(classes);
  const emptyProps: IHabitCreateNameProps = {
    classes,
  };
  const sampleProps: IHabitCreateNameProps = {
    classes,
  };
  return { classes, classesKeys, emptyProps, sampleProps };
};

describe('HabitCreateName component', () => {
  it('match snapshot', () => {
    const { sampleProps } = testSetup();
    const component = shallow(<HabitCreateName {...sampleProps} />);
    expect(component.debug()).toMatchSnapshot();
  });

  test.todo('sample placeholder');
});

describe('HabitCreateName styles', () => {
  it('contains principle classes', () => {
    const { classesKeys } = testSetup();
    expect(habitCreateNameStyles).toBeDefined();
    classesKeys.forEach((keyValue) => {
      // @ts-ignore
      expect(habitCreateNameStyles[keyValue]).toBeDefined();
    });
  });
});

describe('HabitCreateName decorators', () => {
  it('provides styled classes from decorators', () => {
    const { sampleProps, classesKeys } = testSetup();
    const component = mount(<HabitCreateNameDecorated {...sampleProps} />);
    // @ts-ignore
    const assignedClasses = component.find('HabitCreateName').props().classes;
    // @ts-ignore
    classesKeys.forEach(keyValue => {
      expect(assignedClasses[keyValue]).toBeDefined();
    });
  });
});
