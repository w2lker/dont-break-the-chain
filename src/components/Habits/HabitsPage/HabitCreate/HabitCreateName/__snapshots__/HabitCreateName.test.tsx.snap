// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HabitCreateName component match snapshot 1`] = `
"<div className=\\"habitCreateName-wrapper\\">
  HabitCreateName Component
</div>"
`;
